### PR TITLE
Bugfix/arbitrary self types#4138

### LINKED ISSUE #4138: "Implement members" ignores arbitrary self types.
## Environment

* **IntelliJ Rust plugin version:** 0.2.102.2682-192-nightly

## Problem description

`self` is inserted as `Self` instead of one of a few new self types stabilized recently on the way to arbitrary self types.

## Steps to reproduce

Code:
```rust
trait X {
    fn x(self: Pin<&mut Self>);
    fn y(self: *const Self);
}

struct A{}

impl X for A {
}
```
Use the "implement members" quick fix.

Screenshot_20190712_164425

### PR DESCRIPTION
**Pull request by fan-tom**
Fixes #4138

### CODE REVIEW

**Comment by fan-tom** [comment]
How do I limit tests to not run on old_rust? I tried MinRustcVersion annotation but it seems it doesn't work

**Comment by Undin** [comment]
>  I tried MinRustcVersion annotation but it seems it doesn't work

And it's expected because this annotation is supported only by `RsWithToolchainTestBase` (as described in javadoc)

But indeed, there are cases when this annotation may be useful in `RsTestBase` too. So after #4154 it will be possible

File: src/test/kotlin/org/rust/ide/refactoring/implementMembers/ImplementMembersHandlerTest.kt [hunk 1]
```diff
@@ -8,9 +8,13 @@ package org.rust.ide.refactoring.implementMembers
 import com.intellij.openapi.actionSystem.ex.ActionManagerEx
 import junit.framework.TestCase
 import org.intellij.lang.annotations.Language
+import org.rust.ProjectDescriptor
 import org.rust.RsTestBase
+import org.rust.WithStdlibRustProjectDescriptor
+import org.rust.cargo.MinRustcVersion
 import org.rust.ide.inspections.RsTraitImplementationInspection
 
+@ProjectDescriptor(WithStdlibRustProjectDescriptor::class)
```
Comment (Undin) [inline src/test/kotlin/org/rust/ide/refactoring/implementMembers/ImplementMembersHandlerTest.kt]: Let's avoid setting up `WithStdlibRustProjectDescriptor` for whole test class if not all test cases require stdlib. It slows down tests without real need

**Comment by vlad20012** [comment]
Could you, please, squash commits?

**Pull request merged**